homeassistant/components/vicare/binary_sensor.py

```diff
@@ -151,8 +124,11 @@ def device_class(self):
     def update(self):
         """Update state of sensor."""
         try:
-            self._state = self._sensor[CONF_GETTER](self._api)
+            with catch_not_supported() as self._state:
```

Comment:
What's this `as self._state` doing? 

```suggestion
            with catch_not_supported():
```

Reply:
Good question, initially I thought I might have to return some special state in that case but I guess None is okay if a sensor is not available.
In the future we should somehow detect if sensors are available and only create them in that case.